src/pybind/mgr/dashboard_v2/models/nodb.py

```diff
@@ -0,0 +1,897 @@
+
+    Before, NodbQuerySet called self.model.get_all_objects to generate a list of objects. The
+    implementations of get_all_objects were calling the librados commands to fill all attributes,
+    even if they were never accessed.
+
+    Because a field may never be accessed, this can generate better performance than caching,
+    especially if the cache is cold.
+
+    The bulk_attribute_setter decorator can be used like so:
+    >>> class MyModel(NodbModel):
+    >>>     my_field = models.IntegerField()
+    >>>
+    >>>     @bulk_attribute_setter(['my_field'])
+    >>>     def set_my_field(self, objs, field_names):
+    >>>         self.my_field = 42
+
+    Keep in mind, that you can set the my_field attribute on all objects, not just self.
+
+    The decorator modifies the model to look like this:
+    >>> def set_my_field(self, objs):
+    >>>     self.my_field = 42
+    >>>
+    >>> class MyModel(NodbModel):
+    >>>     my_field = models.IntegerField()
+    >>>     set_my_field = LazyPropertyContributor(['my_field'], set_my_field)
+
+    A LazyPropertyContributor property implements the contribute_to_class method, which modifies
+    the model itself to look like so:
+    >>> class MyModel(NodbModel):
+    >>>     my_field = LazyProperty('my_field', set_my_field)
+
+    The my_field field is not overwritten, because the fields are already moved into the _meta class
+    at this point. If someone then accesses the my_field attribute, LazyProperty.__get__ is called,
+    which then calls set_my_field to set the field, as if one had written:
+    >>> instances = MyModel.objects.all()
+    >>> set_my_field(instances[0], instances)
+    >>> assert instances[0].my_field == 42
+
+    For example, get_all_objects generates a QuerySet like this:
+
+    id	name	  disk_usage
+    0	'foo'     LazyProperty('disk_usage')
+    1	'bar'	  LazyProperty('disk_usage')
+
+    When accessing bar.disk_usage, LazyProperty calls `ceph df` and fills the queryset like so:
+
+    id	name	disk_usage
+    0	'foo'   1MB
+    1	'bar'  	2MB
+
+    :type field_names: list[str]
+    :param catch_exceptions: Exceptions that will be caught. In case of an exception, all
+        `field_names` will be set to None.
+    :type catch_exceptions: exceptions.Exception | tuple[exceptions.Exception]
+    """
+
+    if not len(field_names):
+        raise ValueError('`field_names` must not be empty.')
+
+    class LazyPropertyContributor(object):
+        def __init__(self, field_names, func):
+            self.field_names = field_names
+            self.func = func
+
+        def contribute_to_class(self, cls, name, virtual_only=False):
+            for name in self.field_names:
+                setattr(cls, name, LazyProperty(name, self.func, catch_exceptions,
+                                                self.field_names))
+
+    def decorator(func):
+        return LazyPropertyContributor(field_names, func)
+
+    return decorator
+
```

Comment:
Add another blank line

Reply:
I took care of all linting errors today.